src/Radio/Radio.js

```diff
@@ -56,231 +61,66 @@ type Props = {
-    RadioControl_backgroundColor_checked: colors.checked,
-    RadioControl_backgroundColor_checkedHover: colors.hover,
-
-    RadioControl_borderColor: colors.regular,
-    RadioControl_borderColor_focus: colors.focus,
-    RadioControl_borderColor_hover: colors.hover,
-    RadioControl_borderColor_checked: colors.checked,
-    RadioControl_borderColor_checkedFocus: colors.background,
-    RadioControl_borderColor_checkedHover: colors.hover,
-
-    RadioControl_boxShadow_focus: `0 0 0 1px ${colors.background}, 0 0 0 2px ${colors.focus}`,
-    RadioControl_boxShadow_checkedFocus: `0 0 0 1px ${colors.focus}`,
-
-    RadioControl_marginHorizontal: baseTheme.space_inline_md,
-    RadioControl_size: pxToEm(16),
-    RadioControl_size_jumbo: pxToEm(24),
-
-    RadioIcon_size: pxToEm(8),
-    RadioIcon_size_jumbo: pxToEm(16),
-
-    RadioText_color_text: baseTheme.color_text_80,
-    RadioText_fontSize: baseTheme.fontSize_ui,
-    RadioText_fontSize_small: pxToEm(12),
-
-    ...baseTheme
-  };
-};
-
-const styles = {
-  control: ({ disabled, size, theme: baseTheme }) => {
-    const theme = componentTheme(baseTheme);
-    const rtl = theme.direction === 'rtl';
-    const backgroundColor = disabled
-      ? theme.backgroundColor_disabled
-      : theme.RadioControl_backgroundColor;
-    const controlDimensions =
-      size === 'jumbo'
-        ? theme.RadioControl_size_jumbo
-        : theme.RadioControl_size;
-    const svgDimensions =
-      size === 'jumbo' ? theme.RadioIcon_size_jumbo : theme.RadioIcon_size;
-
-    return {
-      alignItems: 'center',
-      backgroundColor,
-      borderColor: theme.RadioControl_borderColor,
-      borderRadius: '100%',
-      borderStyle: 'solid',
-      borderWidth: '1px',
-      color: backgroundColor,
-      content: ' ',
-      display: 'inline-flex',
-      flex: 'none',
-      height: controlDimensions,
-      justifyContent: 'center',
-      marginLeft: rtl ? theme.RadioControl_marginHorizontal : 0,
-      marginRight: rtl ? 0 : theme.RadioControl_marginHorizontal,
-      width: controlDimensions,
-
-      '& svg': {
-        fill: 'currentColor',
-        height: svgDimensions,
-        width: svgDimensions
-      }
-    };
-  },
-  input: ({ disabled, theme: baseTheme }) => {
-    const theme = componentTheme(baseTheme);
-
-    return {
-      ...hideVisually(),
-
-      // NOTE: These hover styles are only needed for the "states" demo.
-      // Otherwise they are applied from styles.root
-      '&:hover,&[data-simulate-hover]': {
-        '& + span': {
-          borderColor: !disabled && theme.RadioControl_borderColor_hover
-        }
+    ...mapComponentThemes(
+      {
+        name: 'Choice',
+        theme: choiceComponentTheme(baseTheme)
       },
-
-      '&:focus,&[data-simulate-focus]': {
-        '& + span': {
-          boxShadow: theme.RadioControl_boxShadow_focus
+      {
+        name: 'Radio',
+        theme: {
+          RadioControl_borderRadius: '100%'
```

Comment:
The diff makes this hard to read, but holy crap this is lovely.

Reply:
Yay!